src/Talis/Persona/Client/Tokens.php

```diff
@@ -229,21 +229,28 @@ public function isPresignedUrlValid($url,$secret) {
      * @throws \Exception
      */
     public function getUserByGupid($gupid, $token){
-        if(trim($gupid) === '')
+        if(!is_string($gupid) || trim($gupid) === '')
         {
             throw new \InvalidArgumentException("Invalid gupid");
         }
-        if(trim($token) === '')
+        if(!is_string($token) || trim($token) === '')
         {
             throw new \InvalidArgumentException("Invalid token");
         }
         $url = $this->config['persona_host'].'/users/?gupid='.$gupid;
-        $user = $this->personaGetUser($url, $token);
 
-        if(isset($user) && !empty($user))
+        try
         {
-            return $user;
-        } else
+            $user = $this->personaGetUser($url, $token);
+            if(isset($user) && !empty($user))
+            {
+                return $user;
+            } else
+            {
+                throw new \Exception('User profile not found');
+            }
+
+        } catch(\Exception $e)
         {
             throw new \Exception('User profile not found');
```

Comment:
This could hide an exception that you didn't expect. Could you add `$e` as a previous exception to the new exception?
```
public __construct ([ string $message = "" [, int $code = 0 [, Exception $previous = NULL ]]] )
```